spec/lib/jobs/generate-enclosure-spec.js

```diff
@@ -318,6 +329,115 @@ describe(require('path').basename(__filename), function () {
+            var node = {
+                id: nodeId,
+                relations: []
+            };
+            var enclId = uuid.v4();
+            var encl = [
+                {
+                    id: enclId,
+                    name: 'Enclosure Node ABC123',
+                    type: 'enclosure',
+                    relations: []
+                },
+            ];
+
+            var enclRelation = {
+                relations: [{
+                    relationType: 'encloses',
+                    targets: [nodeId]
+                }]
+            };
+
+            this.sandbox.stub(mockWaterline.nodes,'create');
+            this.sandbox.stub(mockWaterline.nodes,'find').resolves(encl);
+            this.sandbox.stub(mockWaterline.nodes,'findByIdentifier').resolves(node);
+            return job.run()
+            .then(function() {
+                expect(mockWaterline.nodes.create).to.not.have.been.called;
+                expect(mockWaterline.nodes.updateByIdentifier)
+                    .to.have.been.calledWith(enclId, enclRelation);
+            });
+        });
+
+        it('Should not override other relations when update node', function() {
+            var node = {
+                id: nodeId,
+                relations: [
+                    {
+                        relationType: 'poweredBy',
+                        targets: 'bbb'
+                    }
+                ]
+            };
+            var enclId = uuid.v4();
+            var encl = [
+                {
+                    id: enclId,
+                    name: 'Enclosure Node ABC123',
+                    type: 'enclosure',
+                    relations: [
+                        {
+                           relationType: 'poweredBy',
+                           targets: 'aaa'
+                        }
+                    ]
+                },
+            ];
+            var enclRelation = _.cloneDeep(encl[0].relations);
+            enclRelation.push({
+                relationType: 'encloses',
+                targets: [nodeId]
+            });
+            var nodeRelation = _.cloneDeep(node.relations);
+            nodeRelation.push({
+                relationType: 'enclosedBy',
+                targets: [enclId]
+            });
+
+            this.sandbox.stub(mockWaterline.nodes,'create');
+            this.sandbox.stub(mockWaterline.nodes,'find').resolves(encl);
+            this.sandbox.stub(mockWaterline.nodes,'findByIdentifier').resolves(node);
+            return job.run()
+            .then(function() {
+                expect(mockWaterline.nodes.create).to.not.have.been.called;
+                expect(mockWaterline.nodes.updateByIdentifier)
+                    .to.have.been.calledWith(enclId, {relations: enclRelation});
+                expect(mockWaterline.nodes.updateByIdentifier)
+                    .to.have.been.calledWith(nodeId, {relations: nodeRelation});
```

Comment:
could you also verify the expected state of the node after this sequence? The test verifies that methods were called or not, but doesn't appear to be verifying that the nodes (and content of those objects) remain unchanged